Fix open/close bugs

hansonmcoombs opened the pull request:
Fix 2 bugs regarding MFList when reading external files.

First, allow the use of pathlib.Path objects rather than just strings

Second, fix a bug where calculating mxact raises an error when reading a stress period file with only one feature.

Created two new tests to support these features and ran all automated tests.  The only failures were:

- autotest.test_mbase.test_resolve_exe_by_abs_path
- autotest.test_mbase.test_resolve_exe_by_name
- autotest.test_example_notebooks.test_notebooks
- autotest.regression.test_mf6.test006_2models_different_dis
- autotest.regression.test_mf6.test006_create_tests_2models_gnc

From the traceback I belive that these failures are based on my configuration and setup not the changes (note  that many of these are not related to Modflow2005/nwt

hansonmcoombs committed:
fix(): bugfix in open/close in MFList

fix bug where open/close fails on one cell in a SPD and support pathlib.Path in MFlist. New tests added.

No breaking change

hansonmcoombs committed:
fix(): bugfix in open/close in MFList

fix bug where open/close fails on one cell in a SPD and support pathlib.Path in MFlist. New tests added.

No breaking change

mwtoews commented on flopy/utils/util_list.py:
```diff
@@ -343,7 +349,7 @@ def __cast_data(self, data):
                 elif isinstance(d, int):
                     self.__cast_int(kper, d)
-                elif isinstance(d, str):
+                elif isinstance(d, str) or isinstance(d, Path):
```
```suggestion
                elif isinstance(d, (str, PathLike)):
```

mwtoews commented on flopy/utils/util_list.py:
```diff
@@ -10,6 +10,7 @@
 
 import os
 import warnings
+from pathlib import Path
```
```suggestion
from os import PathLike
```

mwtoews commented on autotest/test_util_2d_and_3d.py:
```diff
@@ -614,3 +614,40 @@ def test_mflist_fromfile(function_tmpdir):
         assert flx_array[k, i, j] == flx
+
+
+def test_pathlib_support():
```
Use the `tmp_path` fixture:
```suggestion
def test_pathlib_support(tmp_path):
```
then follow guidelines [documented here](https://docs.pytest.org/en/latest/how-to/tmp_path.html).

mwtoews commented on autotest/test_util_2d_and_3d.py:
```diff
@@ -614,3 +614,40 @@ def test_mflist_fromfile(function_tmpdir):
+        m.write_input()
+
+
+def test_open_close_one_val():
```
Similar to previous:
```suggestion
def test_open_close_one_val(tmp_path):
```
then rewrite to use fixture.

mwtoews reviewed (commented):
Howdy Matt, I've provided a few suggestions.

hansonmcoombs committed:
Update flopy/utils/util_list.py

Co-authored-by: Mike Taves <mwtoews@gmail.com>

hansonmcoombs committed:
Update flopy/utils/util_list.py

Co-authored-by: Mike Taves <mwtoews@gmail.com>

hansonmcoombs committed:
fix(): implement mwtoews suggestions

fix bug where open/close fails on one cell in a SPD and support pathlib.Path in MFlist. New tests added.

No breaking change

hansonmcoombs commented:
@mwtoews sorted
